### PR TITLE
Smarter SigmaMap handling and needed_parameters

### PR DESCRIPTION
**Pull request by xzh19980906**
This PR modifies the logic of SigmaMap. Now the value of it can be
  1. `"file_name"`. Then no scale factor is needed.
  2. `["median_file_name", "upper_file_name", "lower_file_name"]`. The scale factor is needed in parameters, with "_sigma" suffix as before.
  3. `["median_file_name", "upper_file_name", "lower_file_name", "scaler_name"]`

`needed_parameters` in component is also refined. Removed unnecessary parameters.

### CODE REVIEW

**Comment by xzh19980906** [comment]
Any idea why the doc fails? :) @dachengx

File: appletree/config.py [hunk 1]
```diff
@@ -287,60 +290,82 @@ class SigmaMap(Config):
...
-            except KeyError:
-                mesg = (
-                    f"You specified {self.name} as a dictionary. "
-                    f"The key of it should be the name of one "
-                    f"of the likelihood, but it is {llh_name}."
-                )
-                raise ValueError(mesg)
-        else:
-            self._configs = _configs
+        self._configs = self.get_configs()
```
Comment (dachengx) [inline appletree/config.py:293]: @xzh19980906 I think we do not need to set `_configs` as an attribute of `SigmaMap`.

File: appletree/config.py [hunk 2]
```diff
@@ -287,60 +290,82 @@ class SigmaMap(Config):
...
 
         maps = dict()
         sigmas = ["median", "lower", "upper"]
         for i, sigma in enumerate(sigmas):
-            maps[sigma] = Map(name=self.name + f"_{sigma}", default=self._configs_default[i])
+            if isinstance(self._configs_default, list):
+                maps[sigma] = Map(name=self.name + f"_{sigma}", default=self._configs_default[i])
+            else:
+                # If only one file is given, then use the same file for all sigmas
+                maps[sigma] = Map(name=self.name + f"_{sigma}", default=self._configs_default)
```
Comment (dachengx) [inline appletree/config.py:304]: @xzh19980906 I guess here we can find a way to make `SigmaMap` fallback to `Map`. I can think about that.

Comment (dachengx) [reply appletree/config.py:304]: Anyway, I don't really need to be that clever.

**Review by dachengx — APPROVED** [review]

**Pull request merged**